### ISSUE TITLE
[version:0.6.10+hotfix.1] iOS error: 'WebRTC/WebRTC.h' file not found

### ISSUE THREAD

**Issue by aki25** [flutter-webrtc/flutter-webrtc#713]
When i try to build release on ios (Product -> archive), the build fails with `'WebRTC/WebRTC.h'` file not found error.
When i downgrade the dependency to 0.6.7, it works without errors, but i need the changes in the latest version for improved iOS audio handling.
<img width="1680" alt="Screen Shot 2021-10-04 at 4 15 20 AM" src="https://user-images.githubusercontent.com/8652210/135850252-63108631-0f9f-4b15-b60e-b7d1d3f6fc9d.png">

version that fails: `0.6.10+hotfix.1`
version that works: `0.6.7`

Oh and another thing that i noticed while running `pod update` is that `flutter_webrtc.podspec` still states `s.version = '0.2.2'` for almost all the versions, it gets confusing sometimes to check if xcode is using the right version dependency or not 😅

**Comment by viplifes** [comment]
We have the same error on 0.6.10+hotfix.1

**Comment by sm2017** [comment]
`0.6.8` is OK and `0.6.9` introduce the bug

**Comment by chao-camect** [comment]
It doesn't work for me in xcode neither. I am using xcode 13.

**Comment by alexantropoff** [comment]
WebRTC-SDK doesn't have armv7 support, so just remove armv7 from project's architectures or try to rebuild WebRTC-SDK with armv7 support.

**Comment by alexantropoff** [comment]
> Starting from iOS 14, only apple devices with arm64 architecture have been supported, and the last armv7 device is iPhone 5/5s

5s is 64bit :-)

**Comment by devplanet-dp** [comment]
Please exclude `armv7` from flutter_webrtc, WebRTCSDK, and Project and set Build Active Architectures only to 'YES'. It will work.

**Comment by sm2017** [comment]
> @cloudwebrtc I want to support iPhone 5/5s what must I do?
> 
> > try to rebuild WebRTC-SDK with armv7 support.
> 
> @alexantropoff can you explain how?

@cloudwebrtc Can you help me? I want to support iPhone 5/5s and keep update my flutter-webrtc

**Comment by cloudwebrtc** [comment]
hey @sm2017 I am trying to use [github actions](https://github.com/webrtc-sdk/webrtc-build/issues/1) to compile binaries for all platforms. At that time, I will add armv7 to the compilation step. This will solve your problem. If you are in a hurry, you can use the [following steps](https://github.com/webrtc-sdk/webrtc-build/blob/main/docs/build.md#ios-arm) to compile locally

**Comment by nambui2000k** [comment]
> hey @sm2017 I am trying to use [github actions](https://github.com/webrtc-sdk/webrtc-build/issues/1) to compile binaries for all platforms. At that time, I will add armv7 to the compilation step. This will solve your issue. If you are in a hurry, you can use the [following steps](https://github.com/webrtc-sdk/webrtc-build/blob/main/docs/build.md#ios-arm) to compile locally

when is it planned?

**Comment by stlr00** [comment]
Any news?

**Comment by santoshakil** [comment]
I fixed this issue by doing these following steps:

   1. Delete ios folder
   2. Go to your project directory in terminal and run these commands
   3. `flutter create -i objc .`
   4. `flutter clean && flutter pub get`
   5. `cd ios`
   6. `arch -x86_64 pod update`
   7. `arch -x86_64 pod install`
   8. then at the top of your Pod file, paste this line `platform :ios, '10.0'`
   9. right click on ios folder and open in xcode and then set all deployment target to `10`
   
IMPORTANT: I think webrtc is not working on IOS simulator. I'm not sure. Your can run on real device.

**Comment by aytunch** [comment]
With the suggested workarounds it works in real device but there is no way to run it in iOS simulators.

Do you have any suggestions for that?
There are a lot of packages which depend on WebRTC please fix this issue.

**Comment by http600** [comment]
> 



> I fixed this issue by doing these following steps:
> 
> 1. Delete ios folder
> 2. Go to your project directory in terminal and run these commands
> 3. `flutter create -i objc .`
> 4. `flutter clean && flutter pub get`
> 5. `cd ios`
> 6. `arch -x86_64 pod update`
> 7. `arch -x86_64 pod install`
> 8. then at the top of your Pod file, paste this line `platform :ios, '10.0'`
> 9. right click on ios folder and open in xcode and then set all deployment target to `10`
> 
> IMPORTANT: I think webrtc is not working on IOS simulator. I'm not sure. Your can run on real device.

It looks nice to me, would you mind to explain what it does, and why it works, It's better to know the consequence before typing those command to me, thanks.

**Comment by aytunch** [comment]
@http600 does it work with iOS simulator also?

**Comment by sm2017** [comment]
@cloudwebrtc Did you know why I get the error while I'm run application via android studio for iOS but not in xcode?

**Comment by http600** [comment]
> @http600 does it work with iOS simulator also? And can we make it run with armv7 devices using latest WebRTC? (iphone5 and 5s)

I'm still trying

**Comment by codingride** [comment]
I've been trying all day to make this work until I found my workaround. My solution works fine with until this day.
I've added these few lines in my podfile and worked for me so far in the simulator. Not sure if there other problems on the way but at least I was able to run the simulator.

This is how my podfile look at the end of it:
![Screen Shot 2021-12-25 at 9 26 25 PM](https://user-images.githubusercontent.com/7677815/147391337-bba676c3-0e09-437a-989e-52b90cdedb26.png)

**Comment by ShahanAhmedDev** [comment]
> I've been trying all day to make this work until I found my workaround. My solution works fine with until this day. I've added these few lines in my podfile and worked for me so far in the simulator. Not sure if there other problems on the way but at least I was able to run the simulator.
> 
> This is how my podfile look at the end of it: ![Screen Shot 2021-12-25 at 9 26 25 PM](https://user-images.githubusercontent.com/7677815/147391337-bba676c3-0e09-437a-989e-52b90cdedb26.png)

Thank you! that did the trick for me.

**Comment by Adrian1110** [comment]
> I've been trying all day to make this work until I found my workaround. My solution works fine with until this day. I've added these few lines in my podfile and worked for me so far in the simulator. Not sure if there other problems on the way but at least I was able to run the simulator.
> 
> This is how my podfile look at the end of it: ![Screen Shot 2021-12-25 at 9 26 25 PM](https://user-images.githubusercontent.com/7677815/147391337-bba676c3-0e09-437a-989e-52b90cdedb26.png)

Yes, it's working and if you are too lazy to copy from the picture:
`post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['IPHONE_DEPLOYMENT_TARGET']='10.0'
      config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulators*]']='arm64'
      config.build_settings['ONLY_ACTIVE_ARCH']='YES'
    end
  end
end`

**Comment by mafreud** [comment]
worked fine on my physical device.
Any progress?

**Comment by ShahanAhmedDev** [comment]
For the time being it's only working in physical devices but not in (IOS) emulators.

**Comment by mafreud** [comment]
@ShahanAhmedDev ok😊

**Comment by pro100andrey** [comment]
Any updates ?

**Comment by cloudwebrtc** [comment]
You can try the latest version 0.8.3

**Comment by simonbengtsson** [comment]
The only thing that does not work for me with the latest version 0.8.5 is installing on iOS simulator. After setting `ONLY_ACTIVE_ARCH` as suggested above it works though.

Complete post_install hook for reference:

```ruby
# ./ios/Podfile
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
  
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|

      # https://github.com/flutter-webrtc/flutter-webrtc/issues/713
      if target.name == "flutter_webrtc" || target.name == "WebRTC-SDK"
        config.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'
      end
    end
  end
end
```

**Comment by simonbengtsson** [comment]
Verified that the above post_install hook (ONLY_ACTIVE_ARCH = YES) is still needed for builds to succeed on iOS simulators using `flutter 3.0.1` stable and `flutter-webrtc 0.8.7`.

**Comment by ycherniavskyi** [comment]
Let's assume that the issue is resolved. Fore more info/discussion welcome to released Q&A - #1026.
